src/routes/changeRequest.ts

```diff
@@ -86,33 +113,24 @@ export const authorizeChangeRequest = async (req, res) => {
           ],
         });
       }
+
+      await assignAuthorizationToken(person);
     }
+  }
 
-    await assignAuthorizationToken(person);
-    return res.status(201).send({
-      id: changeRequestId,
-      status: "CONFIRMATION_REQUIRED",
-      updated_at: new Date().toISOString(),
-    });
+  if (deliveryMethod === DeliveryMethod.DEVICE_SIGNING) {
+    response.string_to_sign = Date.now().toString();
   }
 
-  return res.status(404).send({
-    errors: [
-      {
-        id: Date.now().toString(),
-        status: 401,
-        code: "invalid_token",
-        title: "Invalid Token",
-        detail: "Token is invalid",
-      },
-    ],
-  });
+  return res.status(201).send(response);
 };
 
 export const confirmChangeRequest = async (req, res) => {
   const { change_request_id: changeRequestId } = req.params;
-  const { person_id: personId, tan } = req.body;
-  const person = await getPerson(personId);
+  const { person_id: personId, tan, device_id: deviceId } = req.body;
```

Comment:
in case of present `device_id`, I guess we should also check string `signature` presence?